File: ideas/procpp/mixin_handler/tests/http.cpp
#include "ReactHandler.h"

int main()
{
    HttpClientServer server(8082);

    server.on<Http::ON_REQUEST>(Utils::to_function([&server](HttpRequest& req, HttpResponse& resp) {
        resp.body = "content from c++\r\n";
    }));

    server.event_loop();

    return 0;
}

File: ideas/procpp/mixin_handler/tests/http.php
<?php
require(__DIR__ . "/../build/php/ReactHandler.php");
class PhpCallback extends HttpCallback {
    function run($req, $resp)
    {
        $resp->body = "content from php";
    }
}

$callback = new PhpCallback();


$s=new HttpServer(8082);
$s->on('request', $callback);
$s->set_worker_num(4);
$s->event_loop();

